src/zenml/models/pipeline_run_models.py

```diff
@@ -317,10 +298,7 @@ class PipelineRunRequestModel(
     """Pipeline run model with user, workspace, pipeline, and stack as UUIDs."""
 
     id: UUID
-    stack: Optional[UUID]  # Might become None if the stack is deleted.
-    pipeline: Optional[UUID]  # Unlisted runs have this as None.
-    build: Optional[UUID]
-    deployment: Optional[UUID]
+    deployment: "PipelineDeploymentResponseModel"
```

Comment:
Again not a fan of requiring a response model as input for the request model - shouldn't the deployment ID be enough?

Reply:
Changed it according to our discussion.